### PR TITLE
Make kyuubi spark extension compatible with Spark3.4

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

The main change is copy code from `kyuubi-extension-spark-common_2.12` and `kyuubi-extension-spark-3-3`. The reason copy `kyuubi-extension-spark-common_2.12` is that, Spark3.4 v1 writes does not have ctas...

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-extension-spark-3-4/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSparkSQL.g4 [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+whereClause
+    : WHERE booleanExpression
+    ;
+
+zorderClause
+    : ZORDER BY order+=multipartIdentifier (',' order+=multipartIdentifier)*
+    ;
+
+booleanExpression
```
Comment (cfmcgrady) [inline extensions/spark/kyuubi-extension-spark-3-4/src/main/antlr4/org/apache/kyuubi/sql/KyuubiSparkSQL.g4:65]: we don't need `booleanExpression` anymore after https://github.com/apache/kyuubi/pull/2893, let's rebase the master branch

**Comment by cfmcgrady** [comment]
we need to enable the GA job for the spark 3.4 extension too.

https://github.com/apache/kyuubi/blob/fb20bb1f0ca56e1e34a5fd971dd8076b09f18b1d/.github/workflows/style.yml#L37

**Comment by cfmcgrady** [comment]
> we need to enable the GA job for the spark 3.4 extension too.
> 
> https://github.com/apache/kyuubi/blob/fb20bb1f0ca56e1e34a5fd971dd8076b09f18b1d/.github/workflows/style.yml#L37

and `License check` job

https://github.com/apache/kyuubi/blob/fb20bb1f0ca56e1e34a5fd971dd8076b09f18b1d/.github/workflows/license.yml#L45-L48

**Comment by cfmcgrady** [comment]
also cc @wForget @cxzl25

**Review by cfmcgrady — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks for review, merging to master

**Pull request closed without merging**